apps/nestjs-wallet/docs/tutorial.md

```diff
@@ -1,5 +1,56 @@
 # Tutorial
 
+## Business overview
+The business objective of this tutorial is to demonstrate how to issue a proof in the form of a Verifiable Credential (VC) to a known user connected to a portal that we will call "authority portal".
+In the example below, we will issue a permanent residency card to a user we will call "citizen". We use the context "https://w3id.org/citizenship/v1" which describes the data of a VC of type "PermanentResidentCard".
+
+Here is an extract of the context:
+
+        ...
+        "id": "@id",
+        "type": "@type",
+
+        "ctzn": "https://w3id.org/citizenship#",
+        "schema": "http://schema.org/",
+        "xsd": "http://www.w3.org/2001/XMLSchema#",
+
+        "birthCountry": "ctzn:birthCountry",
+        "birthDate": {"@id": "schema:birthDate", "@type": "xsd:dateTime"},
+        "commuterClassification": "ctzn:commuterClassification",
+        "familyName": "schema:familyName",
+        "gender": "schema:gender",
+        "givenName": "schema:givenName",
+        "lprCategory": "ctzn:lprCategory",
+        "lprNumber": "ctzn:lprNumber",
+        "residentSince": {"@id": "ctzn:residentSince", "@type": "xsd:dateTime"}
+        ...
+
+To get the definition of the field "gender" which points to the definition "schema:gender", use the URI of "schema" which points to "http://schema.org/", so we will have a definition available at "http://schema.org/gender".
+TODO interpretation of id and type.
+
+We therefore assume that the citizen is connected to the autority portal and that the information contained in the context about the citizen is available in the autority portal database.
+
+Business workflow:
+- The citizen requests a "PermanentResidentCard" proof on the autority portal
+- The autority portal displays a QR code for the citizen to scan with his mobile wallet.
+- The citizen's mobile wallet contacts the autority portal to prove that he has a DID
+- The citizen's mobile wallet contacts the autority portal again to receive the "PermanentResidentCard"
+- The autority portal responds by sending the "PermanentResidentCard" proof containing the citizen's information signed by the autority portal.
+
+## Technical overview
+From a technical point of view, in this tutorial, we have access to the server API but no mobile wallet is available. So we will use the server API for the portal and the mobile wallet.
+
+We will follow the following steps in this tutorial:
+- [Authority portal action] Create a DID for the autority portal in the form of a key DID method
+- [Authority portal action] Register the DID as default DID for the authority portal
+- [Authority portal action] List the available proofs that the autority portal can issue, an other tutorial is dedicated to demonstrate how to add new available proof for the authority portal
+- [Authority portal action] Create a VC request url that can be transmitted to the citizen as a QR code or a deep link
```

Comment:
These are the same action I believe. Currently `GET Credential Offer` returns both the available Credential types and the url to start the workflow to request the credentials.
```
{
    "typeAvailable": "PermanentResidentCard",
    "vcRequestUrl": "http://localhost:3000/elia-issuer/start-workflow/permanent-resident-card"
}
```
My suggestion would be to modify as shown.
```suggestion
- [Authority portal action] Create a "credential offer" which lists the available credential types that the autority portal can issue and a VC request url that be used to start a request flow. This credential offer can be transmitted to the citizen as a QR code or a deep link.
```

Reply:
But when you get the vcRequestUrl, you need to make a new action to get the : 
`{
    "errors": [],
    "vpRequest": {
        "challenge": "57ca126c-acbf-4da4-8f79-447150e93128",
        "query": [
            {
                "type": "DIDAuth"
            }
        ],
        "interact": {
            "service": [
                {
                    "type": "VcApiPresentationService2021",
                    "serviceEndpoint": "http://localhost:3000/elia-issuer/active-flows/c4e0e642-02a2-4a1a-91c6-40069b328238"
                }
            ]
        }
    }
}`
This will be transformed in a qr code.
So it seems to be two actions to the server api, one to know what is available and the second to get the content of the qr code

Reply:
Ah gotcha. Yes, I'm not sure whether or not the "mobile wallet" would consume the `Credential Offer` or the `VP Request`. I suppose the `Credential Offer` isn't standardized, so you're probably right that it is better to assume that the mobile wallet would consume the `VP Request` as a QR code. How about this wording?
```suggestion
- [Authority portal action] List the available credentials that the autority portal can issue. The citizen, using the portal, would select a credential that they want to request.
- [Authority portal action] Create a [VP Request](https://w3c-ccg.github.io/vp-request-spec/) that can be transmitted to the citizen as a QR code or a deep link
```